lib/storage.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const data = {};
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'));
+const storage = module.exports = {};
+const dataDir = `${__dirname}/../data`;
+
+storage.setItem = function(name, item){
+  // TODO: check if directory exits
+  return  fs.statAsync(`${dataDir}/${name}`)
+  .catch(err => {
+    err.status = 400;
+    return Promise.reject(err);
+  })
+  .then(() => {
+    let json = JSON.stringify(item);
+    return fs.writeFileAsync(`${dataDir}/${name}/${item.id}.json`, json);
+  })
+  .then(() => item);
+};
+
+storage.getItem = function(name, id){
+  // TODO: read json file parse and send back
+
+  // TODO check if name dir xists if not 404
+  return fs.statAsync(`${dataDir}/${name}/${id}.json`)
+  // TODO check if JSON file with id name exists if not 404
+  .catch(err => {
+    err.status = 404;
+    return Promise.reject(err);
+  })
+  .then(() => {
+    return fs.readFileAsync(`${dataDir}/${name}/${id}.json`);
+  })
+  .then(data => {
+    return JSON.parse(data.toString());
+  });
+
+};
+
+storage.deleteItem = function(name, id){
```

Comment:
This delete method should also be using ``fs`` methods to delete the actual file itself. I won't dock you points for this, but just an FYI. It's good to know how to delete a file as well using the fs module. :)